modules/core/src/v2/coinFactory.ts

```diff
@@ -115,11 +129,15 @@ export class CoinFactory {
     throw new errors.UnsupportedCoinError(name);
   }
 
-  public registerCoinConstructor(name: string, constructor: CoinConstructor) {
+  public registerCoinConstructor(name: string, constructor: CoinConstructor, isTestnetEnvDependant = false): void {
```

Comment:
spelling error here on `dependent` and also drop the `Testnet` from the option name. Also let's use an options object instead of a trailing boolean param.

Reply:
done